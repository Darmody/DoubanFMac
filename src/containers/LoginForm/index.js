// @flow
import React, { PureComponent } from 'react'
import { reduxForm } from 'redux-form'

type Props = {
  component: ReactClass<*>,
  handleSubmit: Function,
}

class LoginFormContainer extends PureComponent {
  props: Props

  render() {
    const Component = this.props.component

    return <Component handleSubmit={this.props.handleSubmit} />
  }
}

export default reduxForm({
  form: 'login'
})(LoginFormContainer)
