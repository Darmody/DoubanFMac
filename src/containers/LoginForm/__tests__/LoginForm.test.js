import React from 'react'
import renderer from 'react-test-renderer'
import LoginForm from 'containers/LoginForm'

describe('Container::LoginForm', () => {
  test('渲染正确的 DOM 结构', () => {
    const mockComponent = jest.fn(() => <form>form</form>)
    const component = renderer.create(
      <LoginForm
        component={mockComponent}
        onSubmit={fn => fn}
      />
    )
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
    expect(mockComponent.mock.calls.length).toBe(1)
    expect(mockComponent.mock.calls[0][0].handleSubmit).toBeInstanceOf(Function)
  })
})
