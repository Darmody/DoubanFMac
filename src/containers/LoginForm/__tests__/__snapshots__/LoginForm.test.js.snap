// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container::LoginForm 渲染正确的 DOM 结构 1`] = `
<form>
  form
</form>
`;
